core/drivers/fsl_sec/hw_key_blob.c

```diff
@@ -217,20 +224,57 @@ TEE_Result get_hw_unq_key_blob_hw(uint8_t *hw_key, int size)
 TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
 {
 	TEE_Result res;
+	uint8_t temp_buffer[KEY_BLOB_SIZE];
 
 	res = init_and_roll_forward_master_key();
 	if (res != TEE_SUCCESS) {
 		EMSG("Failed to init normal priblob otpmk identity from CAAM");
 		return res;
 	}
 
-	res = get_hw_unq_key_blob_hw(&hwkey->data[0], HW_UNIQUE_KEY_LENGTH);
+	// Use the first 16 bytes for the HUK, the last 16 for the die ID.
+	res = get_hw_unq_key_blob_hw(temp_buffer, KEY_BLOB_SIZE);
+	memcpy(&hwkey->data[0], temp_buffer, HW_UNIQUE_KEY_LENGTH);
 
 	if (res != TEE_SUCCESS) {
 		EMSG("Hardware Unique Key Failed");
 	} else {
 		DMSG("Hardware Unique Key Retrieved:");
-		DHEXDUMP(&hwkey->data[0], HW_UNIQUE_KEY_LENGTH);
+	}
+
+	return res;
+}
+
+int tee_otp_get_die_id(uint8_t *buffer, size_t len)
+{
```

Comment:
get_hw_unq_key_blob_hw uses int for size, should we be consistent?

Reply:
Its implementing a weak function (which uses int for some reason). Probably just old OP-TEE code no one has looked at in a while.